modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc

```diff
@@ -0,0 +1,144 @@
+====
+
+.Prerequisites
+
+* You must have permission to perform the following actions:
+** Create build configs and start builds.
+** Discover which `SharedSecret` custom resource (CR) instances are available by entering the `oc get sharedsecrets` command and getting a non-empty list back.
+** Determine if the `builder` service account available to you in your namespace is allowed to use the given `SharedSecret` CR instance. In other words, you can run `oc adm policy who-can use <identifier_of_specific_SharedSecret>` to see if the `builder` service account in your namespace is listed.
+
+[NOTE]
+====
+If neither of the last two prerequisites in this list are met, establish, or ask someone to establish, the necessary role-based access control (RBAC) so that you can discover `SharedSecret` CR instances and enable service accounts to use `SharedSecret` CR instances.
+====
+
+.Procedure
+
+. Create a `SharedSecret` object instance with the cluster's entitlement secret by running the `oc apply` command:
++
+[IMPORTANT]
+====
+You must have cluster administrator permissions to create `SharedSecret` objects.
+====
++
+[source,terminal]
+----
+$ oc apply -f -<<EOF
+apiVersion: sharedresource.openshift.io/v1alpha1
+kind: SharedSecret
+metadata:
+  name: share-rhel-entitlement
+spec:
+  secretRef:
+    name: etc-pki-entitlement
+    namespace: openshift-config-managed
+EOF
+----
+
+. Create a role to grant permission to access the `SharedSecret` object and a `RoleBinding` object that grants permission to access the `SharedSecret` object by running the following command:
++
+[source,terminal]
+----
+$ oc new-project buildah-entitlement
+$ oc apply -f -<<EOF
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: use-shared-secret
+rules:
+  - apiGroups:
+      - sharedresource.openshift.io
+    resources:
+      - sharedsecrets
+    resourceNames:
+      - share-entitlement-secret
+    verbs:
+      - use
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: use-shared-secret
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: use-shared-secret
+subjects:
+  - kind: ServiceAccount
+    name: pipeline
+    namespace: buildah-entitlement
+EOF
+----
+
+. Create a build with relevant mount details using `buildah` buildStrategy:
++
+[source,terminal]
+----
+$ oc apply -f -<<EOF
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: buildah-rhel-entitlement-build
+spec:
+  source:
+    type: Git
+    git:
+      url: <location of operator's organization>
+      revision: quick-tests
+    contextDir: rh-builds
+  strategy:
+    name: buildah-csi
+    kind: ClusterBuildStrategy
+  paramValues:
```

Comment:
@apoorvajagtap I believe we now have a GitHub repository for sample code, and the buildah task has/will be updated to have the appropriate volume defined?

Reply:
Agree that we should add a Git repo in there that contains the example

Reply:
Yes, I informed Shivani about this during our sync-up. She'll be adding the correct reference.
Also, @shivanisathe25 for 1.2 the `strategy.name` should be set to `buildah`. (for 1.1, with the additional step 1, the strategy.name is correct: `buildah-csi`)

```suggestion
    git:
      url: https://github.com/redhat-openshift-builds/samples/
    contextDir: buildah-build
  strategy:
    name: buildah-csi
    kind: ClusterBuildStrategy
  paramValues:
```